pkg/services/rpcsrv/server_test.go

```diff
@@ -4135,10 +4139,10 @@ func BenchmarkHandleIn(b *testing.B) {
 }
 
 func TestFailedPreconditionShutdown(t *testing.T) {
-	_, srv, _ := initClearServerWithCustomConfig(t, func(c *config.Config) {
+	chain, srv, _ := initClearServerWithCustomConfig(t, func(c *config.Config) {
 		c.ApplicationConfiguration.RPC.Addresses = []string{"not an address"}
 	})
-
+	defer chain.Close()
```

Comment:
Good, this should solve the problem. But while we're here I see that the common pattern of `initClearServerWithCustomConfig` usage is to always shutdown both chain and server. And thus, we need to do the following:
1. Move chain and RPC server shutdown to a `t.Cleanup` functions. Move these cleanups to the place where chain and RPC server are started respectively.
2. Remove all deferred chain/RPC server shutdown from the callers of `initClearServerWithCustomConfig` and related methods.
3. In a separate commit always close http server as far (if it was started). This should be done in a separate `t.Cleanup`. And BTW, this is likely the source of problems with leaking fds in tests and network-related tests timeouts like https://github.com/nspcc-dev/neo-go/issues/3300.
4. Review all failing tests from our list that may be closed by 3. Link them with commit from 3.

Reply:
about 3 - waiting for tests, but locally it doesn't help

Reply:
If it doesn't then remove the linked issue and adjust the comment. But we need this commit.